Lib/pydoc.py

```diff
@@ -2247,8 +2247,8 @@ def log_message(self, *args):
 
     class DocServer(http.server.HTTPServer):
 
-        def __init__(self, port, callback):
-            self.host = 'localhost'
+        def __init__(self, host, port, callback):
+            self.host = host
```

Comment:
Ah!  Thanks for noting these classes are inside a private method, that excludes compat considerations.  Patch looks fine.